ext/zend_test/tests/strings_marked_as_utf8.phpt

```diff
@@ -115,6 +115,33 @@ $string = "\xff";
 $string_concat = str_repeat($string, 100);
 var_dump(zend_test_is_string_marked_as_valid_utf8($string_concat));
 
+echo "implode:\n";
+$arr = ['a', 'b'];
+$string_concat = implode('', $arr);
+var_dump(zend_test_is_string_marked_as_valid_utf8($string_concat));
+$string_concat = implode('|', $arr);
+var_dump(zend_test_is_string_marked_as_valid_utf8($string_concat));
+$string_concat = implode('', ['c', ...$arr]);
+var_dump(zend_test_is_string_marked_as_valid_utf8($string_concat));
+$string_concat = implode('', [...$arr, 'c']);
+var_dump(zend_test_is_string_marked_as_valid_utf8($string_concat));
+$string_concat = implode('', ["\xff", ...$arr]);
```

Comment:
```suggestion
$string_concat = implode('', ["\xff", ...$arr]);
```
Might make sense to shove the ``"\xff"`` into a constant and call it something like INVALID_UTF8_CHAR as it always takes me a second to understand what's going on. But that can be done later.

Reply:
why `int` (converted to a string) should to not be marked as a UTF-8 string?

`0-9` digits are perfectly valid UTF-8/ASCII chars

Reply:
The IS_LONG case does not affect the flag, thus the string will remain marked as valid utf8, which is correct.

Reply:
That's not what I'm saying.... I am saying that I'm expecting your current patch to mark the result of the following call to implode as not a valid UTF-8 string.

Obviously digits are valid ASCII characters I even added support to all int to string conversions to mark them as valid UTF-8...

Reply:
test case `var_dump(implode("a", ["HELLO", 5, "WORLD", 9, "WTF"]))` will produce valid UTF-8 string which is expected from this PR

please note the flags are ANDed, thus for INT, we do not have to check if the produced string will be UTF-8 valid, as it always be